### ISSUE TITLE
[Meta] Logstash Javafication

### ISSUE THREAD

**Issue by andrewvc** [elastic/logstash#8614]
Below is a prioritized list of projects to convert Logstash internals and plugin API interfaces into Java:

- [ ] Java Plugin API @jordansissel 
- [ ] Java Input API
- [ ] Java Processor API (filters+outputs)
- [ ] Configuration API
- [ ] Runtime Plugin Loading https://github.com/elastic/logstash/issues/8375
- [ ] Pipeline compile-to-java @original-brownbear 
- [ ] Structured Logging in log4j2 @yaauie 
- [ ] Java entrypoint @original-brownbear 
- [ ] Queue multiple-write @colinsurprenant 
  - [ ] Impacts Java Input API
- [ ] Port Rake jobs to Gradle
- [x] Core specs to be executed from JUnit @original-brownbear 
- [ ] (Porting core specs is out of scope for this)
- [ ] Java plugin publishing tooling (blocked on dependency: Java API)
- [ ] Setting YAML to Java
- [ ] Move to snake_yaml, at minimum
- [ ] Grammar (really low priority)

**Comment by original-brownbear** [comment]
@andrewvc Let's tick the "Pipeline compile-to-java" section here? As discussed the differential gain from going further here is kinda small. I guess we can make some incremental improvements to this every now and then, but big gains won't come of just improving the execution.
Java entry point is incoming in #8161 but I'm waiting for #8599 to be a little quicker when ironing out the last details.

**Comment by original-brownbear** [comment]
@colinsurprenant jup, the most performance critical parts of this are compiled at runtime now.

**Comment by colinsurprenant** [comment]
@original-brownbear ok thanks, so yeah I agree on your comment above. Once I am done on the PQ batch maximization I'll resume the multi-write work I started which will/should include include moving the batch readers in Java...

**Comment by kares** [comment]
sounds like all of this is done by now, could we close?
